source/handlers/client.py

```diff
@@ -27,21 +31,37 @@ def __init__(self, sock, clients, ip, port):
         self._port = port
         self._running = True
         self._socket_lock = Lock()
-
         print '[*] %s:%d has connected to the server' % (ip, port)
 
     def run(self):
         while self._running:
             try:
                 request = self._recieve_request()
-                self._socket.send(json.dumps(request))
+                request = Request.deserialize(request)
+                self.handle_request(request.code, request.args)
+
             except ValueError:
                 self._send_bad_request()
             except socket.timeout:
                 continue
             except socket.error:
                 return self.close()
 
+
+    def handle_request(self,code, args):
+        try:
+            controller_function = get_controller_func(code)
+            
+            if controller_function:
+                response = get_controller_func(code)(args)
```

Comment:
There is no need to call this function twice. You can just use the `collector_function` you have already received.